cypress/e2e/registration.cy.js

```diff
@@ -1,11 +1,36 @@
 /// <reference types='cypress' />
 
-describe('Student Registration page', () => {
-  before(() => {
-
-  });
-
-  it('', () => {
-
+describe('Automation Practice Form', () => {
+  it('should fill all fields except picture and submit the form', () => {
+    cy.visit('https://demoqa.com/automation-practice-form');
+    cy.get('#firstName').type('John');
+    cy.get('#lastName').type('Doe');
+    cy.get('#userEmail').type('john.doe@example.com');
+    cy.get('label[for="gender-radio-1"]').click();
+    cy.get('#userNumber').type('1234567890');
+    cy.get('#dateOfBirthInput').click();
+    cy.get('.react-datepicker__month-select').select('April');
+    cy.get('.react-datepicker__year-select').select('1990');
+    cy.get('.react-datepicker__day--015').click();
+    cy.get('#subjectsInput').type('Math{enter}');
+    cy.get('label[for="hobbies-checkbox-1"]').click(); // Sports
+    cy.get('label[for="hobbies-checkbox-2"]').click(); // Reading
+    cy.get('#currentAddress').type('123 Main St, Springfield');
+    cy.get('#state').click();
+    cy.get('#react-select-3-option-0').click();
+    cy.get('#city').click();
+    cy.get('#react-select-4-option-0').click();
+    cy.get('#submit').click();
```

Comment:
After clicking the submit button, it's a good practice to assert that some expected behavior happened, like a success message being displayed or the page being redirected. This way, you can make sure the form was indeed submitted.